### PR TITLE
Add UI tests

### PR DESCRIPTION
**Pull request by uhafner**
Temporary followup for https://github.com/jenkinsci/junit-plugin/pull/342 to see if all tests pass in CI with latest release of junit-plugin.

### CODE REVIEW

**Comment by timja** [comment]
Wrong folder, needs to be under workflows

**Comment by uhafner** [comment]
Oh thanks, that *is* the problem...

File: .github/workflows/ui-tests.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        env:
+          BROWSER: firefox-container
+        run: mvn -V -ntp test --file ui-tests/pom.xml -Dtest=BuildTestResultsTest -Dgpg.skip -Dsurefire.rerunFailingTestsCount=1
+
+  details:
+    runs-on: [ubuntu-latest]
+    name: Details UI tests
+
+    steps:
+      - uses: actions/checkout@v1
```
Comment (timja) [inline .github/workflows/ui-tests.yml:49]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/ui-tests.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
```
Comment (timja) [inline .github/workflows/ui-tests.yml:51]:
```suggestion
        uses: actions/setup-java@v2
```

File: .github/workflows/ui-tests.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        with:
+          java-version: 11
```
Comment (timja) [inline .github/workflows/ui-tests.yml:53]:
```suggestion
          java-version: 11
          distribution: 'temurin'
          cache: 'maven'
```

File: .github/workflows/ui-tests.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  build-results:
+    runs-on: [ubuntu-latest]
+    name: Build results UI tests
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
Comment (timja) [inline .github/workflows/ui-tests.yml:36]: could the version move to a variable?

Comment (uhafner) [reply .github/workflows/ui-tests.yml:36]: Interesting, normally dependabot updates those files as well. I have no idea why not in this place. I will replace that part with the config of the warnings plugin.

**Comment by timja** [comment]
The number of files crashes my browser 😢 

Could you remove the manual caching and switch to adding `cache: maven` in `setup-java`?

**Comment by uhafner** [comment]
I see, there is now separate support for maven! 

I will push my changes to jenkinsci/junit-plugin#342 and make a review there. Then the students see my review comments and can make changes on their own. I will close this PR.

**Comment by timja** [comment]
> I see, there is now separate support for maven!
> 
> I will push my changes to #342 and make a review there. Then the students see my review comments and can make changes on their own. I will close this PR.

sure ❤️

**Comment by timja** [comment]
> I see, there is now separate support for maven!

It was quite error prone before so they switched to including it in the actions themselves

**Pull request closed without merging**